### PR TITLE
Add default index on company for object graph usecases

### PR DESCRIPTION
**Pull request by 0x726f**
Add an index on `company` by default. This helps with most of the existing RQL use-cases where we fetch all records belonging to a company into memory and process them.

### CODE REVIEW

File: mongoengine/base/document.py [hunk 1]
```diff
@@ -930,6 +930,27 @@ def _from_son(cls, son, _auto_dereference=True, only_fields=None, created=False,
...
+    def _build_rippling_company_indices(cls):
+        """Add (company) as an index if required."""
+        res = []
+        if cls._meta.get('no_company_index', False):
+            return res
+
+        ReferenceField = _import_class('ReferenceField')
+        if "company" in cls._fields and isinstance(cls._fields["company"], ReferenceField):
+            if "_cls" in cls._fields and cls._meta.get('allow_inheritance', ALLOW_INHERITANCE):
+                res.append({
```
Comment (Chipe1) [inline mongoengine/base/document.py:943]: `list_indexes()` method in mongoengine/document.py should also be modified to include these indexes. To test whether it works, the output of `SomeDocClass.compare_indexes()` and `SomeDocClass.compare_indexes_including_options()` should be empty

File: mongoengine/base/document.py [hunk 2]
```diff
@@ -930,6 +930,27 @@ def _from_son(cls, son, _auto_dereference=True, only_fields=None, created=False,
 
         return obj
 
+    @classmethod
+    def _build_rippling_company_indices(cls):
+        """Add (company) as an index if required."""
+        res = []
+        if cls._meta.get('no_company_index', False):
```
Comment (Chipe1) [inline mongoengine/base/document.py:937]: Should also exclude this for history models? See the check in `_unique_with_indexes()`

**Comment by Chipe1** [comment]
Please create a [ci full] PR in rippling-main with this mongoengine version. There are some index tests in that repo

File: mongoengine/base/document.py (hunk 1 above)
Comment (0x726f) [reply mongoengine/base/document.py:943]: @Chipe1 I think this is good as it is. 

I ran `<Model>.list_indexes()` while testing this with rippling-main yesterday and saw the new indexes that are being created here. The output of this function is stored in <Model>._meta["index_spec"] from where all of the other functions you mentioned get their index information.

File: mongoengine/base/document.py (hunk 2 above)
Comment (0x726f) [reply mongoengine/base/document.py:937]: Good point! RQL queries don't operate on history tables. I'll push another patch now.

**Review by Chipe1 — APPROVED** [review]
Let's wait for green rippling-main build before merging

**Review by Chipe1 — APPROVED** [review]

**Pull request merged**